tests/integration/SdkTest_test.cpp

```diff
@@ -4065,3 +4069,57 @@ TEST_F(SdkTest, DISABLED_SdkGetRegisteredContacts)
     ASSERT_EQ(js2, std::get<2>(table[1])); // ud
 }
 
+
+bool buildLocalFolders(fs::path targetfolder, const string& prefix, int n, int recurselevel, int filesperfolder)
+{
+    fs::path p = targetfolder / fs::u8path(prefix);
+    if (!fs::create_directory(p))
+        return false;
+
+    for (int i = 0; i < filesperfolder; ++i)
+    {
+        string filename = "file" + to_string(i) + "_" + prefix;
+        fs::path fp = p / fs::u8path(filename);
+#if (__cplusplus >= 201700L)
+        ofstream fs(fp/*, ios::binary*/);
+#else
+        ofstream fs(fp.u8string()/*, ios::binary*/);
+#endif
+        fs << filename;
+    }
+
+    if (recurselevel > 0)
+    {
+        for (int i = 0; i < n; ++i)
+        {
+            if (!buildLocalFolders(p, prefix + "_" + to_string(i), n, recurselevel - 1, filesperfolder))
+                return false;
+        }
+    }
+
+    return true;
+}
+
+TEST_F(SdkTest, RecursiveUploadWithLogout)
+{
+    // this one used to cause a double-delete
+
+    // make new folders (and files) in the local filesystem - approx 90 
+    fs::path p = fs::current_path() / "uploadme_mega_auto_test_sdk";
+    if (fs::exists(p))
+    {
+        fs::remove_all(p);
+    }
+    fs::create_directories(p);
```

Comment:
I've added cleanup code - the problem is that if the test fails, the cleanup doesn't happen, so it needs to pre-cleanup in case the last run failed.  thanks